src/devices/Mcp3xxx/Mcp30xx32xx.cs

```diff
@@ -0,0 +1,205 @@
+        /// <param name="adcResolutionBits">The number of bits of resolution for the ADC.</param>
+        public Mcp30xx32xx(SpiDevice spiDevice, byte pinCount, byte adcResolutionBits) : base(spiDevice)
+        {
+            PinCount = pinCount;
+            _adcResolutionBits = adcResolutionBits;
+        }
+
+        /// <summary>
+        /// Checks that the channel is in range of the available channels pins and throws an exception if not.
+        /// </summary>
+        /// <param name="channel">Channel to be checked</param>
+        /// <param name="channelCount">Value representing the number of channels on the device which may vary depending on the configuration.</param>
+        protected void CheckChannelRange(int channel, int channelCount)
+        {
+            if (channel < 0 || channel > channelCount - 1)
+            {
+                throw new ArgumentOutOfRangeException($"ADC channel must be within the range 0-{channelCount - 1}.", nameof(channel));
+            }
+        }
+
+        /// <summary>
+        /// Checks that the channel is in range of the available input pins and that both pins are part of a valid pairing of
+        /// input pins.
+        /// </summary>
+        /// <param name="positiveChannel">Positive channel to be checked</param>
+        /// <param name="negativeChannel">Negative channel to be checked</param>
+        /// <param name="channelCount">Value representing the number of channels on the device which may vary depending on the configuration.
+        /// If channelCount is not set then the PinCount is used.</param>
+        protected void CheckChannelPairing(int positiveChannel, int negativeChannel, int channelCount = -1)
+        {
+            CheckChannelRange(positiveChannel, (channelCount == -1) ? PinCount : channelCount);
+            CheckChannelRange(negativeChannel, (channelCount == -1) ? PinCount : channelCount);
+
+            // Check that the channel pins are adjacent and part of the same differential pairing
+            if (positiveChannel / 2 != negativeChannel / 2 || positiveChannel == negativeChannel)
+            {
+                throw new ArgumentException($"ADC differential channels must be different and part of the same channel pairing.", nameof(positiveChannel) + " " + nameof(negativeChannel));
+            }
+        }
+
+        /// <summary>
+        /// Reads a  value from the device using pseudo-differential inputs
+        /// </summary>
+        /// <param name="positiveChannel">Channel which represents the signal (valid values: 0 to pincount - 1).</param>
+        /// <param name="negativeChannel">Channel which represents the signal ground (valid values: 0 to pincount - 1).</param>
+        /// <returns>A value corresponding to relative voltage level on specified device channels</returns>
+        public virtual int ReadPseudoDifferential(int positiveChannel, int negativeChannel)
```

Comment:
I'd prefer these to be called `referenceChannel` and `valueChannel` (or something along these lines) as I think this would be more clear what's relative to what

Reply:
I have done this and I sort of agree that the names are better however on all of the datasheets (not exhaustive but including other ADC manufacturers) they are described in the terms positive and negative.